### ISSUE TITLE
Don't count `type: ignore` pragmas towards `line-too-long`

### ISSUE THREAD

**Issue by nickdrozd** [pylint-dev/pylint#8378]
### Current problem

Sometimes a `type: ignore` comment is required for Mypy, but then that increases the line length to the point of triggering Pylint `line-too-long` warning. Then a `pylint: disable = line-too-long` comment needs to be added, which is annoying.

### Desired solution

In this silly example, the second line shouldn't trigger `line-too-long` (with `max-line-length = 72`):

```python
REALLY_LONG_VARIABLE_NAME_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX: int
REALLY_LONG_VARIABLE_NAME_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX = 'string'  # type: ignore
```

**Comment by mbyrnepr2** [comment]
This could perhaps be considered a duplicate of https://github.com/PyCQA/pylint/issues/4802 given the similarity of the situation?

**Comment by jacobtylerwalls** [comment]
Thanks for pointing to #4802. I think it's distinct, though. #4802 hashed out whether to change the behavior that ignores pylint pragmas for line-too-long, and we'll close it with a docs change. This one is a code change: to alter the string search to look for additional pragma statements.

**Comment by nickdrozd** [comment]
Ruff does this, so no reason to pursue it here.

**Comment by DanielNoord** [comment]
@nickdrozd I don't understand? Why wouldn't we also implement this behaviour?

**Comment by Pierre-Sassoulas** [comment]
Opened a more generic issue in #10172

**Comment by nickdrozd** [comment]
If it's easy to do, then sure. But since this issue has been open for almost two years, I'm assuming that it isn't easy to do (also https://github.com/pylint-dev/pylint/issues/8201). And with Black and Ruff around, Pylint's formatting checks are pretty much obsolete. Pylint's own pylintrc file disables them entirely. Of course everyone is free to work on whatever they want, but to me high effort + low payoff = not gonna happen.

**Comment by DanielNoord** [comment]
I can understand that you'd want to clean up your own "Opened issues" tab, but to me this issue served as a good reminder on the `pylint` tracker of a feature we would want to build. If you're not interested in building it, somebody else might :)
